cli/read.go

```diff
@@ -33,6 +33,156 @@ import (
+		fmt.Printf(" Invalid provides section: %s\n", err.Error())
+	} else if len(provides) == 0 {
+		fmt.Printf(" {}\n")
+	} else {
+		fmt.Printf("\n")
+		providesKeys := sortedKeys(provides)
+		for _, provideKey := range providesKeys {
+			fmt.Printf("%s%s: %s\n", strings.Repeat(defaultIndentation, indentationLevel+1), provideKey, (provides)[provideKey])
+		}
+	}
+}
+
+func printDepends(p handlers.Installer, indentationLevel int) {
+	depends, err := p.GetUpdateDepends()
+	fmt.Printf("%sDepends:", strings.Repeat(defaultIndentation, indentationLevel))
+	if err != nil {
+		fmt.Printf(" Invalid depends section: %s\n", err.Error())
+	} else if len(depends) == 0 {
+		fmt.Printf(" {}\n")
+	} else {
+		fmt.Printf("\n")
+		dependsKeys := sortedKeys(depends)
+		for _, dependKey := range dependsKeys {
+			fmt.Printf("%s%s: %s\n", strings.Repeat(defaultIndentation, indentationLevel+1), dependKey, (depends)[dependKey])
+		}
+	}
+}
+func printClearsProvides(p handlers.Installer, indentationLevel int) {
+	caps := p.GetUpdateClearsProvides()
+	fmt.Printf("%sClears Provides:", strings.Repeat(defaultIndentation, indentationLevel))
+	if len(caps) == 0 {
+		fmt.Printf(" []\n")
+	} else {
+		fmt.Printf("\n")
+		for _, cap := range caps {
+			fmt.Printf("%s- %s\n", strings.Repeat(defaultIndentation, indentationLevel+1), cap)
+		}
+	}
+}
+
+func printUpdateMetadata(p handlers.Installer, indentationLevel int) {
+	metaData, err := p.GetUpdateMetaData()
+	fmt.Printf("%sMetadata:", strings.Repeat(defaultIndentation, indentationLevel))
+	if err != nil {
+		fmt.Printf(" Invalid metadata section: %s\n", err.Error())
+	} else if len(metaData) == 0 {
+		fmt.Printf(" {}\n")
+	} else {
+		var metaDataSlice []byte
+		if err == nil {
+			metaDataSlice, err = json.Marshal(metaData)
+		}
+		var metaDataBuf bytes.Buffer
+		if err == nil {
+			err = json.Indent(&metaDataBuf, metaDataSlice, strings.Repeat(defaultIndentation, indentationLevel+1), defaultIndentation)
+		}
+		if err != nil {
+			fmt.Printf(" Invalid metadata section: %s\n", err.Error())
+		} else {
+			fmt.Printf("\n")
+			fmt.Printf("%s%s\n", strings.Repeat(defaultIndentation, indentationLevel+1), metaDataBuf.String())
+		}
+	}
+}
+
+func printPayload(p handlers.Installer, indentationLevel int) {
+	updateType := p.GetUpdateType()
+	if updateType == nil {
+		emptyType := "Empty type"
+		updateType = &emptyType
+	}
+	fmt.Printf("%sType: %v\n", strings.Repeat(defaultIndentation, indentationLevel), *updateType)
+
+	printProvides(p, indentationLevel)
+	printDepends(p, indentationLevel)
+	printClearsProvides(p, indentationLevel)
+	printUpdateMetadata(p, indentationLevel)
+	printFiles(p.GetUpdateAllFiles(), indentationLevel)
+}
+
+func printUpdates(updatePayloads map[int]handlers.Installer, indentationLevel int) {
+	fmt.Printf("%sUpdates:\n", strings.Repeat(defaultIndentation, indentationLevel))
+	for k, payload := range updatePayloads {
+		fmt.Printf("%sUpdate-%d:\n", strings.Repeat(defaultIndentation, indentationLevel+1), k)
```

Comment:
(relates to the previous comment).

Here I suggest either a list of maps, as suggested for `Files`, or to use strings like `0000`, `0001`, etc to align with the Artifact protocol definition.

No solution is perfect, but both are better that introducing a prefix `Update-` IMO.

Reply:
There used to be "space prefixed integers", but these would disturb the formatting, I'll change align it with your excellent `Files` suggestion!